internal/service/appstream/user_test.go

```diff
@@ -0,0 +1,230 @@
+)
+
+func TestAccAppStreamUser_basic(t *testing.T) {
+	var userOutput appstream.User
+	resourceName := "aws_appstream_user.test"
+	authType := "USERPOOL"
+	domain := acctest.RandomDomainName()
+	rEmail := acctest.RandomEmailAddress(domain)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(t)
+			acctest.PreCheckHasIAMRole(t, "AmazonAppStreamServiceAccess")
+		},
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckUserDestroy,
+		ErrorCheck:        acctest.ErrorCheck(t, appstream.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUserConfig(authType, rEmail),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckUserExists(resourceName, &userOutput),
+					resource.TestCheckResourceAttr(resourceName, "authentication_type", authType),
+					resource.TestCheckResourceAttr(resourceName, "user_name", rEmail),
+					acctest.CheckResourceAttrRFC3339(resourceName, "created_time"),
+					resource.TestCheckResourceAttr(resourceName, "enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "status", "FORCE_CHANGE_PASSWORD"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAppStreamUser_disappears(t *testing.T) {
+	var userOutput appstream.User
+	resourceName := "aws_appstream_user.test"
+	authType := "USERPOOL"
+	domain := acctest.RandomDomainName()
+	rEmail := acctest.RandomEmailAddress(domain)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(t)
+			acctest.PreCheckHasIAMRole(t, "AmazonAppStreamServiceAccess")
+		},
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckUserDestroy,
+		ErrorCheck:        acctest.ErrorCheck(t, appstream.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUserConfig(authType, rEmail),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckUserExists(resourceName, &userOutput),
+					acctest.CheckResourceDisappears(acctest.Provider, tfappstream.ResourceUser(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAppStreamUser_complete(t *testing.T) {
+	var userOutput appstream.User
+	resourceName := "aws_appstream_user.test"
+	authType := "USERPOOL"
+	firstName := "John"
+	lastName := "Doe"
+	domain := acctest.RandomDomainName()
+	rEmail := acctest.RandomEmailAddress(domain)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acctest.PreCheck(t)
+			acctest.PreCheckHasIAMRole(t, "AmazonAppStreamServiceAccess")
+		},
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckUserDestroy,
+		ErrorCheck:        acctest.ErrorCheck(t, appstream.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccUserCompleteConfig(authType, rEmail, firstName, lastName, true),
```

Comment:
To test creating a user with `enabled = false`, we could set it to `false` for the first one, then set it to `true`, and then a final step that sets it back to `false` to validate that `enabled` can be swapped in both directions